astropy/cosmology/parameters.py

```diff
@@ -150,4 +144,5 @@
                "Table 1 (WMAP + BAO + SN ML).")
 )
 
-available = tuple(k for k in locals() if not k.startswith('_'))
+# If new parameters are added, this list must be updated
```

Comment:
This seems reasonable but I worry that it might be a little unreliable. What is the chance of randomly having another `dict` in your namespace with a member named `reference`? Maybe it's not huge but it doesn't seem impossible. 

It almost seems like each of these should be an instance of a class like `cosmology.Parameter` or something like that (which inherits `dict`), and then the test would be as simple as:
```
available = [k for k in locals() if isinstance(k, Parameter)]
```
But maybe that's too much of a change?